### DISCUSSION TITLE
InlineCSS and other transformers for partial templates?

### DISCUSSION THREAD

**Discussion by m-lotze** [maizzle/framework#741] [Q&A]
We have a template structure like this:

```
partials/h1.html
partials/h2.html
...
default.html
````

with a `config.production.js`:

```
build: {
    templates: {
      destination: {
        path: 'src/views',
        extension: 'blade.php'
      }
    }
  },
````

After running `npm run build` only css and classes in `default.blade.php` are inlined. `h1.blade.php` is created in `src/views/partials' but it is not transformed at all:

```
<h1 class="font-heading text-3xl text-primary-500">{{ $slot }}</h1>
```

How do we achieve a result like this for a partial?

```
<h1 style="font-family: 'Heading', ui-sans-serif, system-ui, -apple-system, 'Segoe UI', sans-serif; font-size: 30px; color: #91192e">{{ $slot }}</h1>
```

It seems like only full HTML files are transformed.

**Comment by cossssmin** [comment]
Are you trying to use `partials/h1.html` as a [component](https://maizzle.com/docs/components) in a template? If so just place it under `src/components`

**Reply by m-lotze** [reply]
No. We need `partials/h1.html` rendered as `partials/h1.blade.php` to use it as a blade component in laravel.

**Reply by cossssmin** [reply]
Oh ok, so it needs to be a template as far as Maizzle is concerned.

You need to have it extend a basic layout that includes `<style>{{{ page.css }}}</style>` - that needs to exist in the HTML in order for the inliner to work. For example:

```html
<!-- partials/h1.html -->
<extends src="layouts/basic.html">
  <block name="template">
    <h1 class="text-3xl">@{{ $slot }}</h1>
  </block>
</extends>
```

```html
<!-- layouts/basic.html -->
<style>{{{ page.css }}}</style>
<block name="template"></block>
```

That will output:

```html
<h1 style="font-size: 30px">{{ $slot }}</h1>
```

However if you have any pseudos or media queries like `sm:text-sm` on the `<h1>`, the `<style>` tag will be removed if you're missing the `<body>` tag - so you need to wrap the template block in a `<body>` tag:

```html
<!-- layouts/basic.html -->
<style>{{{ page.css }}}</style>
<body>
  <block name="template"></block>
</body>
```

Of course, you'll then need to get rid of the `<body>` tag for the final component, which you can do in `afterBuild`:

```js
// config.production.js
module.exports = {
  // ...,
  events: {
    afterTransformers(html) {
      return html.replace(/<\/?body.*?>/g, '')
    },
  },
}
```
